src/pages/zh-cn/core-concepts/endpoints.md

```diff
@@ -0,0 +1,279 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: API 端点
+description: 了解如何创建服务于任何类型数据的 API 端点
+---
+
+Astro 允许你创建自定义的 [API 端点](https://www.cloudflare.com/zh-cn/learning/security/api/what-is-api-endpoint/)（以下简称为“端点”）来提供任何类型的数据。你可以使用它来生成图像、公开 RSS 文档或将它们用作 API 路由来为你的站点构建完整的 API。
+
+在静态生成的站点中，你的自定义端点在构建时被调用以生成静态文件。如果你选择启用 [SSR](/zh-cn/guides/server-side-rendering/) 模式，自定义端点会变成根据请求调用的实时服务器端点。静态和 SSR 端点的定义类似，但 SSR 端点支持附加额外的功能。
+
+## 静态文件端点
+
+要想要创建自定义端点，请将 `.js` 或 `.ts` 文件添加到 `/pages` 目录。`.js` 或 `.ts` 这样的扩展名将在构建过程中被删除，因此文件名中应包含你要创建的数据的扩展名。例如，文件 `src/pages/data.json.ts` 将被构建为 API 端点 `/data.json`。
+
+端点导出一个 `get` 函数（可选的以 `async` 导出），它的唯一参数是一个对象，其具有两个属性（`params` 和 `request`）。它返回一个带有 `body` 的对象，Astro 将在构建时调用它并使用 `body` 中的内容来生成文件。
+
+```js
+// Example: src/pages/builtwith.json.ts
+// Outputs: /builtwith.json
+export async function get({params, request}) {
+  return {
+    body: JSON.stringify({
+      name: 'Astro',
+      url: 'https://astro.build/',
+    }),
+  };
+}
+```
+
+返回对象还可以具有 `encoding` 属性。它可以是任何可以被 Node.js 的 `fs.writeFile` 方法接收的 [`BufferEncoding`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bdd02508ddb5eebcf701fdb8ffd6e84eabf47885/types/node/buffer.d.ts#L169)。例如，想要生成一个二进制的 PNG 图像：
+
+```ts title="src/pages/astro-logo.png.ts" {6}
+export async function get({ params, request }) => {
+  const response = await fetch("https://astro.build/assets/press/full-logo-light.png");
+  const buffer = Buffer.from(await response.arrayBuffer());
+  return {
+    body: buffer,
+    encoding: 'binary',
+  };
```

Comment:
```suggestion
  };
}
```

Reply:
It seems that this error also appears in the [English version](https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L38).

Reply:
> It seems that this error also appears in the [English version](https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L38).

Yes. I think we can change it in this PR.